odoo/addons/base/models/ir_actions_report.py

```diff
@@ -517,10 +517,28 @@ def barcode(self, barcode_type, value, width=600, height=100, humanreadable=0, q
             if barcode_type == 'QR' and quiet:
                 bar_border = 0
 
-            barcode = createBarcodeDrawing(
-                barcode_type, value=value, format='png', width=width, height=height,
-                humanReadable=humanreadable, quiet=quiet, barBorder=bar_border
-            )
+            kwargs = {
+                'value': value,
+                'format': 'png',
+                'width': width,
+                'height': height,
+                'humanReadable': humanreadable,
+                'quiet': quiet,
+                'barBorder': bar_border,
+            }
+
+            # The QR code can have different layouts depending on the Error Correction Level
+            # See: https://en.wikipedia.org/wiki/QR_code#Error_correction
+            # Level 'L' – up to 7% damage   (default)
+            # Level 'M' – up to 15% damage  (i.e. required by l10n_ch QR bill)
+            # Level 'Q' – up to 25% damage
+            # Level 'H' – up to 30% damage
+            if barcode_type == 'QR' and qr_level:
```

Comment:
The span of a `try/except` shoudl be limited unless absolutely necessary, there's a per-barcode-type processing section at the top, this should be added to it (and the `bar_border` stuff folded into as well).

Reply:
You're right, I'll move it above once I get the picture of what we want. 

Also, the `except` is **sus**, I don't think that a wrong QR code should result in a Code128 drawing in case of `ValueError` or `AttributeError`